src/test/java/io/tiledb/java/api/ArrayTest.java

```diff
@@ -209,23 +206,25 @@ public void testArrayOpenAt() throws Exception {
 
     long[] array_a = new long[] {1, 2, 3, 6};
     insertArbitraryValues(new NativeArray(ctx, array_a, Long.class));
-    long ts_a = new Timestamp(System.currentTimeMillis()).toInstant().toEpochMilli();
-
+    Thread.sleep(1000);
```

Comment:
I don't understand why these tests are failing, but TileDB now supports writing with `openedAt`. This unit test might be simpler and fail less if we use that to write and read at fixed times. If a `insertArbitraryValuesAt(NativeArray nativeArray, BigInteger timestamp)` is added you can write the 3 values at a fixed timestamp and then read them at a fixed timestamp. The timestamps can be any time, even 0, 1, 2 should work. Would you mind giving this a try? If the test still fails after this, I'll let @gsvic look into it more.

Reply:
Managed to get things passing with waiting a second before and after the timestamps but I can look at that as an option tomorrow.

Reply:
Modification for that style now included in the PR. Builds passing.

Reply:
Thanks for making the change